javaparser-core/src/main/java/com/github/javaparser/ast/validator/chunks/ModifierValidator.java

```diff
@@ -49,7 +49,7 @@ public void visit(ClassOrInterfaceDeclaration n, ProblemReporter reporter) {
 
     private void validateClassModifiers(ClassOrInterfaceDeclaration n, ProblemReporter reporter) {
         if (n.isTopLevelType()) {
-            validateModifiers(n, reporter, PUBLIC, ABSTRACT, FINAL, STRICTFP);
+            validateModifiers(n, reporter, PUBLIC, ABSTRACT, STATIC, FINAL, STRICTFP);
```

Comment:
Example of the nested static class written in top-level in the Checker Framework
https://github.com/Bohdankm22/checker-framework/blob/master/framework/src/org/checkerframework/common/basetype/flow.astub#L103